### DISCUSSION TITLE
after power failure surprising states

### DISCUSSION THREAD

**Discussion by EDSTOBI** [arendst/Tasmota#15389] [Q&A] [answered]
Good day and happy easter
I need your help because I am a complete beginner with tasmoat shelly 1pm.
I have 4 different pumps wher I monitor with 4 shelly the current, this is the main task,. This is simple and works.
Where I have trouble with are the switch, this switches Shelly 1PM, The Shelly is always on power, with the switch then the pump is switched on via the Shelly.
All pumps have an on / off switch, which is operated periodically. On ist power permanently 240V on SW Shelly terminal  until switch is turned to off,  SW goes from 240V to 0V, pump is shut down.
I thought this would work until I had a power failure. 
After I have restored the power supply,  the 4 Shelly were absolute messed up. I went and turned all the switches to off, and then set all the shelly ofer the web to off via toggel.

My question is how do I avoid this next time power fails
and how can I see in MQTT or in the web API if the pump switch is on over mecanical switch or if someone has switched the pump via the web interface. That woud saves me a lot of legwork

The switch is determining.
Actually I would say at power up always Shelly terminal for Pump is off, unless the external switch is on.
The external switch is the main control source

How do I implement this, or what is provided by default for tasmota Shelly 1PM?

THX
EDSTOBI

**Comment by barbudor** [comment, marked as answer]
I'm not sure to understand your description. This is where a wiring diagram would be useful.

Regarding:
> After I have restored the power supply, the 4 Shelly were absolute messed up.
If by messed up you mean that they lost their settings, then the reason might be the recovery procedure has been triggered because the power got off and on multiple times. Read this : https://tasmota.github.io/docs/Device-Recovery/#fast-power-cycle-device-recovery

> how can I see in MQTT or in the web API if the pump switch is on over mecanical switch or if someone has switched the pump via the web interface.
You can see in the web interface if the power is on or off
You can monitor the MQTT POWER messages to know when it is switched on or off
With WebLog 3 you can see in the Web console the source of a command (web, switch, console, mqtt...) but this is visible only there if WebLog 3 was enabled before the event. And this information is only visible in the console and not sent outside of Tasmota.

> Actually I would say at power up always Shelly terminal for Pump is off, unless the external switch is on.
You can set `PowerOnState OFF` so the Shelly always starts OFF
And use a boot rule to sense the state of the switch such as 
```
Rule1 
  ON switch1#boot DO backlog Status 10 ENDON
  ON StatusSNS#Switch1 DO Power %value% ENDON
```

Hope I answered your questions

**Reply by sfromis** [reply]
`on Switch1#boot do var1 %value% endon` also allows capturing the switch state, but the value is not directly usable for the Power command.

**Comment by EDSTOBI** [comment]
HI Barbudor
May i ask you to explain the rule to me?
I do not finde out what it is dooing, because I do not find the commands.

Sorry but I am a Tasmota beginner

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Rules/

**Reply by EDSTOBI** [reply]
THX  sfromis
I know this page, unfortunately it does not help me with **status 10** and **statusSNS**, so I ask for an explanation.
I also read and search with Google, unfortunately, the search gives me nothing, may be that here the language problem also plays a role, it is not my mother language
edstobi

**Reply by barbudor** [reply]
Beside that fromis provided you a better rule than mine, the explanation of my solution is
ON switch1#boot  : trigger that occur on boot when the Switch comment is ready

  DO backlog Status 10 ENDON : execute the command Status 10 which emit the MQTT Message StatusSNS similar to the tele/../SENSOR message but with an additional JSON level `StatusSNS`
To see the MQTT message sent by Status 10 just type the command in the web console and you'll see
Trick : Allthough DO Status 10 ENDON would have also executed Status 10, having backlog insure that the message out of Status 10 will be processed by the rules engine

ON StatusSNS#Switch1 : Triggers on a command response which include the StatusSNS key and Switch1 subkey. %value% will take the value of subkey.
In essence : RESULT { "StatusSNS": { ...., "Switch1": "ON", ... }

DO Power %value% ENDON : excecute the command Power with the value got from the trigger, ie the value of Switch1
